### PR TITLE
Extend ObjectStore integration tests

### PR DESCRIPTION
**Pull request by VJalili**
This PR does the following: 

1. moves all ObjectStore integration tests, and their related files, to a dedicated package;
2. adds a test asserting the content of the files that ObjectStore creates as a result of running the test tool, i.e., `create_10`. Ideally, this test should be implemented as a separate test method; however, with the current setup of the integration test, test methods within a test case are not executed independently. For instance, with `n` test method in a test case each producing `10` datasets running `create_10` tool, ObjectStore will have `10n` datasets in the output because all the test method use one instance of Galaxy with a common ObjectStore setup. It may worth it to work on making sure there is one Galaxy instance with independent setup per test method of a test class that can be initialized and cleaned-up using `setUp` and `tearDown` methods.

### CODE REVIEW

**Comment by VJalili** [comment]
@jmchilton could you please have a look at this?

File: test/integration/objectstore/test_jobs.py [hunk 1]
```diff
@@ -71,6 +71,25 @@ def test_tool_simple_constructs(self):
...
+        According to the ObjectStore configuration given in the
+        `DISTRIBUTED_OBJECT_STORE_CONFIG_TEMPLATE` variable, datasets
+        can be stored on three backends, named:
+            -   primary/files1;
+            -   primary/files2;
+            -   secondary/files3.
+
+        Objectstore _randomly_ distributes tools outputs on
+        `primary/files1` and `primary/files2`, and will use
+        `secondary/files3` and both `primary` backends fail.
```
Comment (mvdbeek) [inline test/integration/objectstore/test_jobs.py]:
```suggestion
        `secondary/files3` if both `primary` backends fail.
```

File: test/integration/objectstore/test_jobs.py [hunk 2]
```diff
@@ -79,17 +98,39 @@ def test_tool_simple_constructs(self):
...
         # and not actually testing object store behavior.
         assert (files_1_count <= 11) and (files_2_count <= 11)
         assert (files_1_count >= 0) and (files_2_count >= 0)
 
+        # TODO: ideally the following assertion should be separated in a different test method.
+        contents = []
+        path1_files = _get_datasets_files_in_path(self.files1_path)
+        path2_files = _get_datasets_files_in_path(self.files2_path)
+        path3_files = _get_datasets_files_in_path(self.files3_path)
+        for filename in set().union(path1_files, path2_files, path3_files):
```
Comment (mvdbeek) [inline test/integration/objectstore/test_jobs.py]:
```suggestion
        for filename in path1_files + path2_files + path3_files:
```

File: test/integration/objectstore/test_jobs.py [hunk 3]
```diff
@@ -79,17 +98,39 @@ def test_tool_simple_constructs(self):
...
+            with open(filename) as f:
+                content = f.read().strip('\n').strip('\t')
```
Comment (mvdbeek) [inline test/integration/objectstore/test_jobs.py]:
```suggestion
                content = f.read().strip()
```
I don't see where the tab should be coming from.

Comment (VJalili) [reply test/integration/objectstore/test_jobs.py]: no tab, but better to strip the line break to simplify content assertions.

Comment (mvdbeek) [reply test/integration/objectstore/test_jobs.py]: line break (and tabs for that matter) are considered whitespace, so they are removed by `strip()`:
```
In [1]: s = ''

In [2]: ?s.strip
Docstring:
S.strip([chars]) -> str

Return a copy of the string S with leading and trailing
whitespace removed.
If chars is given and not None, remove characters in chars instead.
Type:      builtin_function_or_method

In [3]: s = '\n\t'

In [4]: s.strip()
Out[4]: ''
```

Comment (VJalili) [reply test/integration/objectstore/test_jobs.py]: Good point, thanks.

**Comment by mvdbeek** [comment]
Thanks @VJalili !

**Comment by VJalili** [comment]
Thanks @mvdbeek

**Pull request merged**